src/condor_gridmanager/azureresource.cpp

```diff
@@ -194,9 +194,7 @@ AzureResource::BatchStatusResult AzureResource::StartBatchStatus() {
 	//
 	std::set<AzureJob*> myJobs;
 	AzureJob * nextJob = NULL;
-	BaseJob *nextBaseJob = NULL;
-	registeredJobs.Rewind();
-	while ( (nextBaseJob = registeredJobs.Next()) ) {
+	for (auto nextBaseJob: registeredJobs) {
```

Comment:
Sure we want nextBaseJob by value, and not by reference?  And in other places later?

Reply:
The list elements are BaseJob*, so in practice it doesn't matter if nextBaseJob is by value or by reference.